drivers/soundwire/bus.c

```diff
@@ -399,6 +408,41 @@ sdw_bwrite_no_pm(struct sdw_bus *bus, u16 dev_num, u32 addr, u8 value)
 	return sdw_transfer(bus, &msg);
 }
 
+int
+sdw_bread_no_pm_unlocked(struct sdw_bus *bus, u16 dev_num, u32 addr)
+{
+	struct sdw_msg msg;
+	u8 buf;
+	int ret;
+
+	ret = sdw_fill_msg(&msg, NULL, addr, 1, dev_num,
+			   SDW_MSG_FLAG_READ, &buf);
+	if (ret)
+		return ret;
+
+	ret = sdw_transfer_unlocked(bus, &msg);
+	if (ret < 0)
+		return ret;
+	else
```

Comment:
not needed

Reply:
it's the same code as in other functions, only with the _unlocked suffix added. If we change the code pattern we change it everywhere or we don't

Reply:
@plbossart I only found 1 more occurrence. #2034 I understand that my nitpicking can be perceived as additional work with "little" benefit and no functional improvement but I think keeping in line with general kernel coding practices at least to some extent is a good idea. I also think we sometimes use the "repeating pattern" argument too much.

Reply:
https://github.com/thesofproject/linux/blob/302fbb0fc8b7d9266aeb9b59144c6fa7fbcb751b/drivers/soundwire/bus.c#L304
https://github.com/thesofproject/linux/blob/302fbb0fc8b7d9266aeb9b59144c6fa7fbcb751b/drivers/soundwire/bus.c#L324
https://github.com/thesofproject/linux/blob/302fbb0fc8b7d9266aeb9b59144c6fa7fbcb751b/drivers/soundwire/bus.c#L393
https://github.com/thesofproject/linux/blob/302fbb0fc8b7d9266aeb9b59144c6fa7fbcb751b/drivers/soundwire/bus.c#L524

Reply:
My point was that this should be cleaned-up separately

Reply:
@plbossart ok, so you don't only mean cases with `sdw_transfer()` but all `if () return x; else return y;` patterns. Sure, unfortunately we have a few of them around, including under drivers/soundwire. But I still don't think they are the *common prevailing* pattern. Even only in drivers/soundwire/bus.c you see both. So, if you count all, I'd say the common pattern is a lack thereof. So we are free to choose either. In which case I'd prefer to go with the kernel-wide preference.

Reply:
I'll extend that PR with a couple more cases, thanks for pointing them out

Reply:
@plbossart since https://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git/commit/drivers/soundwire/bus.c?h=next-20200706&id=f779ad09109eebd3176464fdf6729e3504b49bc5 has now been merged, I think we can remove this `else`

Reply:
done